workspaces/config/lib/definitions/index.js

```diff
@@ -19,10 +19,6 @@ const flatten = (obj, flat = {}) => {
     }
   }
 
-  // XXX make this the bin/npm-cli.js file explicitly instead
-  // otherwise using npm programmatically is a bit of a pain.
-  flat.npmBin = require.main ? require.main.filename
```

Comment:
we still need this set on the flatOptions.  Modules like [`pacote`](https://github.com/npm/pacote/blob/18e760f5c438155c1b8af8aa1ffbead874732058/lib/fetcher.js#L112) use this to spawn npm for scripts

Reply:
Can do. BTW, on flatOptions, the `nodeBin` property is set basically as `process.env.NODE || process.execPath` but in set-envs, the environment variable `$NODE` is set to `process.execPath`.

Do you know why? Is this environment variable supposed to be clobbered?

Reply:
In `set-envs.js` it's set to `config.execPath`.  It's changed by [npm itself](https://github.com/npm/cli/blob/latest/lib/npm.js#L188-L192) if it detects it's running from a symlink